packages/@aws-cdk/core/lib/fs/fingerprint.ts

```diff
@@ -84,6 +96,36 @@ export function fingerprint(fileOrDirectory: string, options: FingerprintOptions
 }
 
 export function contentFingerprint(file: string): string {
+  // On windows it's particularly important to pass bigint: true to ensure that
+  // floating-point inaccuracies don't result in false matches. ( see
+  // https://github.com/nodejs/node/issues/12115#issuecomment-438741212 )
+  //
+  // Note that even if we do get a inode collision somehow, it's unlikely that
+  // both mtime and size would have a false-positive as well.
+
+  // Note: Node introduced statSync's bigint argument around Node 12. We check
+  // here to make sure that we're using node 12 or higher and bypass the cache
+  // otherwise.
+
+  if (Number(process.versions.node.split('.')[0]) < 12) {
+    return contentFingerprintMiss(file);
+  }
+
+  // We also must suppress typescript typechecks as we are using a version of
+  // @types/node that only supports node 10 declarations.
+  // @ts-ignore
+  const stats = fs.statSync(file, { bigint: true });
+  const cacheKey = JSON.stringify({
+    mtime_unix: stats.mtime.getUTCDate(),
+    mtime_ms: stats.mtime.getUTCMilliseconds(),
```

Comment:
This is not doing what is intended:

- `getUTCDate()` returns the UTC day-of-the-month of the Date object
- `getUTCMilliseconds()` returns the UTC milliseconds component of the Date object (i.e: fractional seconds)

Reply:
Thanks for the catch (and fix!)